robosat/tools/train.py

```diff
@@ -179,16 +180,14 @@ def validate(loader, num_classes, device, net, criterion):
         running_loss += loss.item()
 
         for mask, output in zip(masks, outputs):
-            mask = mask.data.cpu().numpy()
-            prediction = output.data.max(0)[1].cpu().numpy()
-            iou.add(mask.ravel(), prediction.ravel())
+            iou.add(mask.float(), output.max(0)[1].float())
```

Comment:
Why do we not have to detach here like above? I don't think I understand how detach/nograd interacts and what the purpose of it is :thinking:

Reply:
No need to detach there because you already have a @no_grad  on validate function

Reply:
:+1: